src/parser/shared/modules/HotTracker.ts

```diff
@@ -393,9 +395,37 @@ abstract class HotTracker extends Analyzer {
       if (!this.hots[targetId]) {
         this.hots[targetId] = {};
       }
-      this.hots[targetId][spellId] = this.bouncingHots[0];
-      this.bouncingHots.shift();
-      return;
+      while (this.bouncingHots.length > 0) {
+        const hot = this.bouncingHots[0];
+        const lastBounce = hot.lastBounce;
+        if (lastBounce) {
+          bounceDebug &&
+            console.log(
+              'Last Bounce: ' +
+                this.owner.formatTimestamp(lastBounce, 3) +
+                ', Event Timestamp: ' +
+                this.owner.formatTimestamp(event.timestamp, 3),
+            );
+          //validate the bounce
+          const timeBetween = event.timestamp - lastBounce;
+          if (timeBetween <= BOUNCE_THRESHOLD) {
+            bounceDebug &&
+              console.log(
+                'Applied a bouncing hot at ' + this.owner.formatTimestamp(event.timestamp, 3),
+              );
+            this.hots[targetId][spellId] = hot;
+            this.bouncingHots.shift();
+            return;
+          } else {
+            bounceDebug &&
+              console.log(
+                'Bouncing Hot lost due to no eligible jump targets, player death, etc' +
+                  this.owner.formatTimestamp(lastBounce, 3),
+              );
+            this.bouncingHots.shift();
+          }
```

Comment:
```suggestion
          }
          bounceDebug &&
            console.log(
              'Bouncing Hot lost due to no eligible jump targets, player death, etc' +
                this.owner.formatTimestamp(lastBounce, 3),
            );
          this.bouncingHots.shift();
```
else is redundant because we return in the if statement